### PR TITLE
Remove PodIP from ContainerStatus

### LINKED ISSUE #5524: Remove PodIP from ContainerStatus
Now kubelet reports PodStatus including PodIP through PodStatus completely, there is no need to have PodIP in ContainerStatus for network container anymore.

I am going to send a pr fixing it shortly. Otherwise, someone free to pick it up.

### PR DESCRIPTION
**Pull request by dchen1107**
Fix #5524

### CODE REVIEW

**Comment by vmarmol** [comment]
Needs a rebase btw.

File: pkg/kubelet/dockertools/docker.go [hunk 1]
```diff
@@ -534,19 +533,29 @@ var (
...
 
-func inspectContainer(client DockerInterface, dockerID, containerName, tPath string) (*api.ContainerStatus, error) {
+// Internal information kept for containers from inspection
+type containerStatusResult struct {
+	status api.ContainerStatus
+	podIP  string
+	err    error
+}
+
+func inspectContainer(client DockerInterface, dockerID, containerName, tPath string) *containerStatusResult {
```
Comment (vmarmol) [inline pkg/kubelet/dockertools/docker.go]: What about returning 3 values rather than the struct?

**Comment by vmarmol** [comment]
Looks good, just the rebase and the nit.

File: pkg/kubelet/dockertools/docker.go (hunk 1 above)
Comment (vishh) [inline pkg/kubelet/dockertools/docker.go]: +1

File: pkg/kubelet/dockertools/docker.go [hunk 2]
```diff
@@ -534,19 +533,29 @@ var (
 	ErrContainerCannotRun = errors.New("Container cannot run")
 )
 
-func inspectContainer(client DockerInterface, dockerID, containerName, tPath string) (*api.ContainerStatus, error) {
+// Internal information kept for containers from inspection
+type containerStatusResult struct {
+	status api.ContainerStatus
+	podIP  string
```
Comment (vishh) [inline pkg/kubelet/dockertools/docker.go]: why is `inspectContainer` returning 'Pod'IP? Why not name it just ip or container ip?

Comment (dchen1107) [inline pkg/kubelet/dockertools/docker.go]: This is indeed PodIP information, and only obtained through network container. We used to expose such information to master through extra container entry called POD in PodStatus. I have cleaned that code by pushing down entire PodStatus to kubelet, and hide POD container completely. This code is goging further to remove PodIP from api.ContainerStatus.

Comment (vishh) [inline pkg/kubelet/dockertools/docker.go]: Dawn I understand the overall intent. But in this specific case, we are inspecting a container and that method is returning a pod IP. That seems weird.

**Comment by dchen1107** [comment]
Rebased.

File: pkg/kubelet/dockertools/docker.go (hunk 1 above)
Comment (dchen1107) [inline pkg/kubelet/dockertools/docker.go]: Too many return values makes code hard to follow.

File: pkg/kubelet/dockertools/docker.go (hunk 2 above)
Comment (dchen1107) [inline pkg/kubelet/dockertools/docker.go]: Good point. Let me change it.

File: pkg/kubelet/dockertools/docker.go (hunk 1 above)
Comment (vmarmol) [inline pkg/kubelet/dockertools/docker.go]: Most of our code takes that route today :) not a blocker for this though.

**Comment by dchen1107** [comment]
Addressed the comments.

File: pkg/kubelet/dockertools/docker.go (hunk 1 above)
Comment (dchen1107) [inline pkg/kubelet/dockertools/docker.go]: I know. Another one I want to change is ParseDockerName :-)

Comment (vmarmol) [inline pkg/kubelet/dockertools/docker.go]: WDYT if we meet in the middle, we have containerStatusResult with a status
and an ip (no error). The function returns (containerStatusResult, error).
I think this is more natural for Go and shouldn't have too many return
types.

File: pkg/kubelet/kubelet.go [hunk 3]
```diff
@@ -2015,44 +2015,45 @@ func (kl *Kubelet) GetPodStatus(podFullName string, uid types.UID) (api.PodStatu
...
 		containerStatus.Ready = kl.readiness.IsReady(containerStatus)
 		podStatus.Info[c.Name] = containerStatus
 	}
 	podStatus.Conditions = append(podStatus.Conditions, getPodReadyCondition(spec, podStatus.Info)...)
 
-	netContainerInfo, found := info[dockertools.PodInfraContainerName]
-	if found {
-		podStatus.PodIP = netContainerInfo.PodIP
-	}
-	podStatus.Host = kl.hostname
```
Comment (vmarmol) [inline pkg/kubelet/kubelet.go]: We stopped populating Host. Was that intended?

**Comment by dchen1107** [comment]
Resolve an issue caused by rebase. Thanks!

**Comment by vmarmol** [comment]
LGTM, will merge on green thanks @dchen1107!
